apps/dcos_overlay/src/dcos_overlay_poller.erl

```diff
@@ -171,19 +173,75 @@ maybe_create_vtep(_Overlay = #mesos_state_agentoverlayinfo{backend = Backend},
     VTEPNameStr = binary_to_list(VTEPName),
     ParsedVTEPMAC = list_to_tuple(parse_vtep_mac(VTEPMAC)),
     {ParsedVTEPIP, PrefixLen} = parse_subnet(VTEPIP),
-    dcos_overlay_netlink:iplink_add(Pid, VTEPNameStr, "vxlan", VNI, 64000),
-    {ok, _} = dcos_overlay_netlink:iplink_set(Pid, ParsedVTEPMAC, VTEPNameStr),
-    {ok, _} = dcos_overlay_netlink:ipaddr_replace(Pid, inet, ParsedVTEPIP, PrefixLen, VTEPNameStr),
-    case {application:get_env(dcos_overlay, enable_ipv6, true), VTEPIP6} of
+    case vtep_present(Pid, VXLan) of
```

Comment:
Could you please move this new code to new function `create_vtep/2`?

Reply:
add vtep link and set mac address can be moved to `create_vtep/2`, but it's better to keep  `vtep_present` here, because it judge where or not should ( maybe ) we create vtep, right?

Reply:
could you please move this `case` expression to new function `maybe_create_vtep_link/2`, rename `create_vtep/2` to `create_vtep_link/2`, and rename `config_l3/2` to `create_vtep_addr/2`?